module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,56 @@
+#include "compression.hpp"
+
+#include <cctype>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedBitmap{};
+
+    for (const auto& el : bitmap) {
+        compressedBitmap.push_back({*begin(el), 1});
+        for (auto iter = std::next(begin(el)); iter != end(el); ++iter) {
+            if (*iter == compressedBitmap.back().first) {
+                compressedBitmap.back().second++;
+            } else {
+                compressedBitmap.push_back({*iter, 1});
```

Comment:
Maybe emplace_back?

Reply:
As far as I know, for built-in types, like pair, there is no difference in efficiency between using `push_back` and `emplace_back`. Still I might be wrong, so we need 3rd person opinion here :)

Reply:
Agree